authors/apps/articles/tests/test_filter_search_functionality.py

```diff
@@ -0,0 +1,152 @@
+from django.urls import reverse
+from rest_framework import status
+from ...authentication.tests.base_class import BaseTest
+
+
+class TestFilterSearchFunctionality(BaseTest):
+    """
+    TestFilterSearchFunctionality handles testing of the search and filter
+    functionality of articles
+    """
+    def setUp(self):
+        super().setUp()
+        self.user = self.activated_user()
+        self.client.force_authenticate(user=self.user)
+        self.article = self.create_article(self.user)
+        self.url = reverse('articles:articles')
+
+    def add_tags_to_article(self):
+        """
+        add_tags_to_article method adds a tag to an article
+        """
+        self.article.tag_list = ["okay"]
+        self.article.save()
+
+    def test_user_can_filter_articles_by_author_username(self):
+        """
+        Test user can filter articles based on author username
+        """
+        username = self.user.username
+        response = self.client.get(f"{self.url}?author={username}")
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(len(response.data), 1)
+        response = self.client.get(f"{self.url}?author=vdhfvsf")
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(len(response.data), 0)
+
+    def test_user_can_filter_by_articles_by_title(self):
```

Comment:
Thanks for the good work done @oma0256. But think you want to do here is name this method as test_user_can_filter_articles_by_title.
Thanks.